core/types/block.libevm.go

```diff
@@ -134,22 +133,11 @@ type BodyHooks interface {
 	RLPFieldPointersForDecoding(*Body) *rlp.Fields
 }
 
-// TestOnlyRegisterBodyHooks is a temporary means of "registering" BodyHooks for
-// the purpose of testing. It will panic if called outside of a test.
-func TestOnlyRegisterBodyHooks(h BodyHooks) {
-	testonly.OrPanic(func() {
-		todoRegisteredBodyHooks = h
-	})
-}
-
-// todoRegisteredBodyHooks is a temporary placeholder for "registering"
-// BodyHooks, before they are included in [RegisterExtras].
-var todoRegisteredBodyHooks BodyHooks = NOOPBodyHooks{}
-
 func (b *Body) hooks() BodyHooks {
-	// TODO(arr4n): when incorporating BodyHooks into [RegisterExtras], the
-	// [todoRegisteredBodyHooks] variable MUST be removed.
-	return todoRegisteredBodyHooks
+	if r := registeredExtras; r.Registered() {
+		return r.Get().hooks.hooksFromBody(b)
+	}
+	return new(NOOPBodyHooks)
```

Comment:
```suggestion
	return NOOPBodyHooks{}
```

(nit) The interface is declared on the value so return that.